commcare_connect/form_receiver/processor.py

```diff
@@ -206,7 +209,12 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
     if flags:
         user_visit.flagged = True
         user_visit.flag_reason = {"flags": flags}
+
+    if opportunity.auto_approve_visits and user_visit.status == VisitValidationStatus.pending:
+        user_visit.status = VisitValidationStatus.rejected if user_visit.flagged else VisitValidationStatus.approved
```

Comment:
I made this comment in the overall PR, but I think we want to auto approve only. Flags stay as pending, thats how we alert users they need to view it.

Reply:
Fix. https://github.com/dimagi/commcare-connect/pull/294/commits/28a898593ab863404c5f4a79fce74f0c6d443491